File: 119-pascals-triangle-ii/pascals-triangle-ii.go
import "fmt"

func getRow(rowIndex int) []int {
	if rowIndex == 0 {
		return []int{1}
	} else if rowIndex == 1 {
		return []int{1, 1}
	} else {
		PreRow := []int{1}
        NextRow := []int{}

        for i:=1; i<=rowIndex; i++ {
		    NextRow = make([]int, i+1)
            for j:=0; j<len(NextRow);j++ {
                fmt.Println(i, j, PreRow, NextRow)
                if j==0 || j == i {
                    NextRow[j] = 1
                } else {
                    NextRow[j] = PreRow[j-1]+PreRow[j]
                }
            }
            PreRow = NextRow
		}
        return NextRow
	}
}

File: 119-pascals-triangle-ii/pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        dp = [[0]*i for i in range(1,rowIndex+2)]
        dp[0][0] = 1
        for i in range(1, rowIndex+1):
            for j in range(i+1):
                # print(i,j)
                if j == 0 or j == i:
                    dp[i][j] = 1
                else:
                    dp[i][j] = dp[i-1][j-1] + dp[i-1][j]
        # print(dp)
        return dp[-1]
        